wangle/service/ServerDispatcher.h

```diff
@@ -94,11 +95,21 @@ class MultiplexServerDispatcher : public HandlerAdapter<Req, Resp> {
       : service_(service) {}
 
   void read(Context* ctx, Req in) override {
-    (*service_)(std::move(in)).then([ctx](Resp resp) {
-      ctx->fireWrite(std::move(resp));
+    std::tuple<Context*, std::shared_ptr<PipelineBase> > guard{ctx, ctx->getPipelineShared()};
+    (*service_)(std::move(in)).then([guard = std::move(guard)](Resp resp) {
+	auto ctx = std::get<0>(guard);
+	auto  pipe = std::get<1>(guard);
+	ctx->fireWrite(std::move(resp));	
+	ctx->getTransport()->getEventBase()->runInEventBaseThread([pipe = std::move(pipe)]{/*nothing but free pipeline in io thread*/});
```

Comment:
Style. 80-column-wide ruler.

This should be a common-enough pattern in IO code or code that moves between IO and CPU threads that a comment may not strictly be necessary. In any event, style-wise, it can go on the previous line.